java/code/src/com/redhat/rhn/frontend/xmlrpc/image/ImageInfoHandler.java

```diff
@@ -175,6 +260,132 @@ else if (StringUtils.isEmpty(storeLabel)) {
         }
     }
 
+    /**
+     * Import an OS image
+     * @param loggedInUser The current user
+     * @param name The name
+     * @param version The version
+     * @param arch The architecture
+     * @return the image id
+     *
+     * @xmlrpc.doc Import an image and schedule an inspect afterwards
+     * @xmlrpc.param #param("string", "sessionKey")
+     * @xmlrpc.param #param_desc("string", "name", "image name as specified in the
+     * store")
+     * @xmlrpc.param #param_desc("string", "version", "version to import")
+     * @xmlrpc.param #param_desc("string", "arch", image architecture")
+     * @xmlrpc.returntype #param_desc("int", "id", "ID of the image")
+     */
+    public Long importOSImage(User loggedInUser, String name, String version, String arch) {
+        ensureImageAdmin(loggedInUser);
+        if (StringUtils.isEmpty(name)) {
+            throw new InvalidParameterException("Image name cannot be empty.");
+        }
+        ImageStore store = ImageStoreFactory.lookupBylabelAndOrg("SUSE Manager OS Image Store", loggedInUser.getOrg())
+                        .orElseThrow(NoSuchImageStoreException::new);
```

Comment:
Not that there is something to do, but I find this quite fragile. IMO lookup by store type label `os_image` would survive if someone will try to do some deep SUMA/Uyuni renames.

Reply:
The same code is used in unit tests so it can't break silently.